### PR TITLE
[PTX-2519] Added timeout to ValidateApplications() API

### PR DESCRIPTION
**Pull request by vinayakshnd**
Jenkins job with px-backup test:
https://jenkins.portworx.co/job/Torpedo/view/Torpedo%20NextPX/job/tp-nextpx-backup/96/console

### CODE REVIEW

File: tests/backup/backup_test.go [hunk 1]
```diff
@@ -187,7 +187,7 @@ var _ = Describe("{BackupCreateKillStoreRestore}", func() {
 			}
 			// TODO(stgleb): Adjust this logic to skip cluster scoped resources
 			// that do not get backed up
-			ValidateApplications(contexts)
```
Comment (stgleb) [inline tests/backup/backup_test.go:190]: Why don't we add timeout to context as a field? Maybe even add context.Context as a field to control timeout and cancellation.

Comment (vinayakshnd) [reply tests/backup/backup_test.go]: Changed as per comment

**Comment by vinayakshnd** [comment]
Jenkins Job with review comments addressed:
https://jenkins.portworx.co/job/tp-backup-vinayak/114/console

**Comment by stgleb** [comment]
Job passed https://jenkins.portworx.co/job/Torpedo/view/Torpedo%20NextPX/job/tp-nextpx-backup/100/

**Pull request merged**